quinn/transformations.py

```diff
@@ -87,28 +88,193 @@ def to_snake_case(s: str) -> str:
+
+                results.extend(
+                    sort_nested_cols(new_struct, is_reversed, base_field=new_base_field)
+                )
+            return results
+
+        def handle_array_type(parent_struct: StructField, is_reversed: bool) -> str:
+            def format_array_selection(
+                elements: list, base_str: str, suffix_str: str
+            ) -> str:
+                element_names = [i.split(".")[-1] for i in elements]
+                array_elements_formatted = [f"x.{i} as {i}" for i in element_names]
+
+                output = (
+                    f"{base_str} {', '.join(array_elements_formatted)} {suffix_str}"
+                )
+                return output
+
+            array_parent = parent_struct.jsonValue()["type"]["elementType"]
+
+            base_str = f"transform({parent_struct.name}"
+            suffix_str = f") AS {parent_struct.name}"
+
+            # if struct in array, create mapping to struct
+            if array_parent["type"] == "struct":
+                array_parent = array_parent["fields"]
+
+                base_str = f"{base_str}, x -> struct("
+                suffix_str = f"){suffix_str}"
+
+            array_elements = parse_fields(array_parent, parent_struct, is_reversed)
+            formatted_array_selection = format_array_selection(
+                array_elements, base_str, suffix_str
+            )
+            return formatted_array_selection
+
+        def handle_struct_type(parent_struct: StructField, is_reversed: bool) -> str:
+            def format_struct_selection(elements: list, struct_name: str) -> str:
+                output: str = f"struct( {', '.join(elements)} ) AS {struct_name}"
+                return output
+
+            field_list = parent_struct.jsonValue()["type"]["fields"]
+            sub_fields = parse_fields(field_list, parent_struct, is_reversed)
+            formatted_sub_fields = format_struct_selection(
+                sub_fields, parent_struct.name
+            )
+            return formatted_sub_fields
+
+        select_cols = []
+        for parent_struct in sorted(schema, key=lambda x: x.name, reverse=is_reversed):
+            field_type = parent_struct.dataType
+            if isinstance(field_type, ArrayType):
+                result = handle_array_type(parent_struct, is_reversed)
+
+            elif isinstance(field_type, StructType):
+                result = handle_struct_type(parent_struct, is_reversed)
+            else:
+                if base_field:
+                    result = f"{base_field}.{parent_struct.name}"
+                else:
+                    result = parent_struct.name
+            select_cols.append(result)
+
+        return select_cols
+
+    def get_original_nullability(field: StructField, result_dict: dict) -> None:
+        def assign_nullability(field: StructField, result_dict: dict) -> dict:
+            try:
+                result_dict[field.name] = field.nullable
+            except AttributeError:
```

Comment:
Is it an antipattern? Why not just to check that `field` has an attribute `nullable` via `hasattr(field, "nullable")`? Generation of stacktrace inside exception is costly operation, if there 1000 fields it may tend to performance degradation.